hack/getting_started.sh

```diff
@@ -0,0 +1,592 @@
+            echo ${promptAfterCmdFail}
+        fi
+        echo "${cmd} failed, exit...."
+        exit 1
+    fi
+
+    if [ -n "${PromptAfterCmdSuc}" ]; then
+        echo ${PromptAfterCmdSuc}
+    fi
+}
+
+function isIPOccupied() {
+    local ip=$1
+    local occupiedIPList=$2
+    echo "Checking if ${ip} is occupied"
+
+    IFS=$'\n'
+    for occupiedIPWithMask in ${occupiedIPList}
+    do
+        occupiedIP=`echo $occupiedIPWithMask | sed -En 's/^(.*)\/([0-9]{1,2})/\1/p'`
+        if [ "${ip}" == "${occupiedIP}" ]; then
+             return 1
+        fi
+    done
+
+    # should not use the gateway address
+    for gateway in ${gateways}
+    do 
+        if [ "${gateway}" == "${ip}" ]; then
+            return 1
+        fi
+    done
+
+    echo "${ip} is not occupied"
+    return 0
+}
+
+
+
+function num2IP() {
+    local num=$1
+    a=$((num>>24))
+    b=$((num>>16&0xff))
+    c=$((num>>8&0xff))
+    d=$((num&0xff))
+    echo "$a.$b.$c.$d"
+}  
+
+function binary2IP() {
+    num=`echo "ibase=2;  $1" | bc`
+    returnValue=$(num2IP $num)
+    echo ${returnValue}
+}
+
+function calcControlPlaneIP() {
+    subNet=`docker network inspect kind | jq -r 'map(.IPAM.Config[].Subnet) []'`
+    gateways=`docker network inspect kind | jq -r 'map(.IPAM.Config[].Gateway) []'`
+    occupiedIPList=`docker network inspect kind | jq -r 'map(.Containers[].IPv4Address) []'`
+
+    for line in ${subNet}
+    do
+        echo ${line} | grep ":"
+        if [ $? -ne 0 ]; then
+            IPMask=${line}
+            break
+        fi
+    done
+
+    echo "IPMask is ${subNet}"
+    echo "occupiedIPList:${occupiedIPList}"
+    echo "gateways:${gateways}"
+
+    ip=`echo $IPMask | sed -En 's/^(.*)\/([0-9]{1,2})/\1/p'`
+    ipSubNetBit=`echo $IPMask | sed -En 's/^(.*)\/([0-9]{1,2})/\2/p'`
+    ipSubHostBit=$[32-${ipSubNetBit}]
+
+    IFS=.
+    hexIP=`for str in ${ip}; do printf "%02X" $str; done`
+    binaryIP=`echo "ibase=16; obase=2; ${hexIP}" | bc`
+
+    ipSubNet=`echo ${binaryIP:0:${ipSubNetBit}}`
+
+    full0="00000000000000000000000000000001"
+    full1="11111111111111111111111111111110"
+
+    minIPBinary=${ipSubNet}`echo ${full0:${ipSubNetBit}}`
+    maxIPBinary=${ipSubNet}`echo ${full1:${ipSubNetBit}}`
+
+    minIPint=`echo "ibase=2; ${minIPBinary}"|bc`
+    maxIPint=`echo "ibase=2; ${maxIPBinary}"|bc`
+
+    for((i=${minIPint};i<=${maxIPint};i++));  
+    do
+        ip=$(num2IP ${i})
+
+        isIPOccupied "${ip}" "${occupiedIPList}" "${gateways}"
+        if [ $? == 0 ]; then
+            controlPlaneEndPointIp=${ip}
+            echo "CONTROL_PLANE_ENDPOINT_IP is ${controlPlaneEndPointIp}"
+            return
+        fi
+    done
+
+    echo "Can't get an available IP for control plane endpoint, exit...."
+    exit 1
+
+}
+
+function installByohProvider() {
+    local writeByoh=0
+    local clusterCtlYamlFile="${HOME}/.cluster-api/clusterctl.yaml"
+    local maxRunTimes=40
+    local waitTime=20
+
+    # Write clusterctl.yaml
```

Comment:
Why is this logic to check if byoh already installed needs to be done? We can directly write the clusterctl.yaml and continue to install the provider. Since we are deleting kind-byoh cluster if it exists and creating a new one, it is safe to assume that this operation will not fail.

Reply:
The file used append mode, we can't directly write the clusterctl.yaml again. 
Are you sure that ${HOME}/.cluster-api/clusterctl.yaml is not possible to created by other?  If so, we could overwrite this file as a fresh one.

Reply:
Question: Do we even need to modify `clusterctl.yaml`, I believe byoh is already one of the supported provider in `clusterctl` via https://github.com/kubernetes-sigs/cluster-api/pull/5593

Reply:
@dharmjit is that supported with CAPI `v1.0.4`?

Reply:
This is introduced in https://github.com/kubernetes-sigs/cluster-api/releases/tag/v1.1.0 and I quickly verified this to be  working. Do we have to specifically use CAPI `v1.0.4`?
```
$ ./clusterctl-linux-amd64 init --infrastructure byoh
Fetching providers
Installing cert-manager Version="v1.5.3"
Waiting for cert-manager to be available...
Installing Provider="cluster-api" Version="v1.1.2" TargetNamespace="capi-system"
Installing Provider="bootstrap-kubeadm" Version="v1.1.2" TargetNamespace="capi-kubeadm-bootstrap-system"
Installing Provider="control-plane-kubeadm" Version="v1.1.2" TargetNamespace="capi-kubeadm-control-plane-system"
I0225 06:45:08.876085  569883 request.go:665] Waited for 1.035277754s due to client-side throttling, not priority and fairness, request: GET:https://127.0.0.1:37561/apis/clusterctl.cluster.x-k8s.io/v1alpha3?timeout=30s
Installing Provider="infrastructure-byoh" Version="v0.1.0" TargetNamespace="byoh-system"

Your management cluster has been initialized successfully!
```

Reply:
@dharmjit Woo, you are right. This part followed getting_started.md, I think we need also to update this document.

Reply:
> Do we have to specifically use CAPI v1.0.4

I see your point now. We can use latest clusterctl in getting started script to avoid this block of code and directly use init, is that right? Makes perfect sense!
I was thinking to do that when rest of the codebase upgrades to latest version of CAPI, but I guess they need not be related.